### PR TITLE
[SDK] Support pfazure init

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

This pull request includes changes to various files mainly related to the `run` method in `src/promptflow-azure/promptflow/azure/_pf_client.py` and tests in `src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py`. The changes involve adding support for `init` in the `run` method, adding a new test case `test_flex_flow_run`, and updating the configuration in `src/promptflow/tests/test_configs/eager_flows/basic_callable_class/flow.dag.yaml`. A minor change was also made to `.cspell.json`.

Here are the most important changes:

Changes to the `run` method:
* [`src/promptflow-azure/promptflow/azure/_pf_client.py`](diffhunk://#diff-bb83ca64a6fa27acbf60fff44e3aa73e56744caf8c326606ba32bd0ddd77b21bL265-L267): The `init` parameter is now supported in the `run` method. The previously raised `NotImplementedError` for `init` has been removed. [[1]](diffhunk://#diff-bb83ca64a6fa27acbf60fff44e3aa73e56744caf8c326606ba32bd0ddd77b21bL265-L267) [[2]](diffhunk://#diff-bb83ca64a6fa27acbf60fff44e3aa73e56744caf8c326606ba32bd0ddd77b21bR301)
* [`src/promptflow-devkit/promptflow/_sdk/entities/_run.py`](diffhunk://#diff-e2a13997b279d2a6b93f4c3e990f08e85dc19896262d5d14354cc294d67362f4R604): The `init_k_wargs` property has been added to the `_to_rest_object` method.

Addition of a new test case:
* [`src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py`](diffhunk://#diff-b63fdccdd1a3a293abe7799c0a231a7226614f726b6de1298f7e0233af26f128R1259-R1287): A new test case `test_flex_flow_run` has been added which tests the run operation with the `init` parameter.

Configuration update:
* [`src/promptflow/tests/test_configs/eager_flows/basic_callable_class/flow.dag.yaml`](diffhunk://#diff-cbdde7417cc8a09c3cc12f4d1406271524ef5efb59d09c81d4dc72540812d6a0R2-R3): The `environment` configuration has been updated with a new `image` property.

Minor change:
* [`.cspell.json`](diffhunk://#diff-4e0f175c840784de9e935f881821905c75a2a125aa57c51268a7fb4496edfbe6L199-R200): The word `wargs` has been added to the list of recognized words.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py [hunk 1]
```diff
@@ -1256,3 +1256,32 @@ def test_eager_flow_run_without_yaml(self, pf: PFClient, randstr: Callable[[str]
 
         # the YAML file will not exist in user's folder
         assert not Path(f"{EAGER_FLOWS_DIR}/simple_without_yaml/flow.dag.yaml").exists()
+
+    def test_flex_flow_run(self, pf: PFClient, randstr: Callable[[str], str]):
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:1260]: no recording?

**Review by 0mza987 — APPROVED** [review]

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/tests/test_configs/eager_flows/basic_callable_class/flow.dag.yaml [hunk 2]
```diff
@@ -1 +1,3 @@
 entry: simple_callable_class:MyFlow
+environment:
+  image: promptflow.azurecr.io/promptflow-runtime:bu-20240403-1452
```
Comment (wangchao1230) [inline src/promptflow/tests/test_configs/eager_flows/basic_callable_class/flow.dag.yaml:3]: is bu short for build?

File: .cspell.json [hunk 3]
```diff
@@ -196,7 +196,8 @@
     "nologo",
     "wscript",
     "raisvc",
-    "evals"
+    "evals",
```
Comment (wangchao1230) [inline .cspell.json:199]: please take care of the conflicts

File: src/promptflow/tests/test_configs/eager_flows/basic_callable_class/flow.dag.yaml (hunk 2 above)
Comment (D-W-) [reply src/promptflow/tests/test_configs/eager_flows/basic_callable_class/flow.dag.yaml]: Not sure, seems auto generated.

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**